lib/internal/mime.js

```diff
@@ -0,0 +1,339 @@
+  return target;
+}
+
+const UNTIL_EQUALS_OR_SEMICOLON = hardenRegExp(/^[^;=]*/);
+const QUOTED_VALUE_PATTERN = hardenRegExp(/^(?:([\\]$)|[\\][\s\S]|[^"])*(?:(")|$)/u);
+const QUOTED_CHARACTER = hardenRegExp(/[\\]([\s\S])/ug);
+function parseParametersString(str) {
+  const paramsMap = new Map();
+  str = StringPrototypeReplace(str, ENDING_WHITESPACE, '');
+  let position = 0;
+  while (position < str.length) {
+    // Skip any whitespace before parameter
+    const skip = execPattern(
+      BEGINNING_WHITESPACE,
+      StringPrototypeSlice(str, position))[0];
+    if (skip) position += skip.length;
+    // Read until ';' or '='
+    const parameterMatch = execPattern(
+      UNTIL_EQUALS_OR_SEMICOLON,
+      StringPrototypeSlice(str, position));
+    const parameterString = toASCIILower(parameterMatch[0]);
+    position += parameterString.length;
+    // If we found a terminating character
+    if (position < str.length) {
+      // Safe to use because we never do special actions for surrogate pairs
+      const char = StringPrototypeCharAt(str, position);
+      // Skip the terminating character
+      position += 1;
+      // Ignore parameters without values
+      if (char === ';') {
+        continue;
+      }
+    }
+    // If we are at end of the string, it cannot have a value
+    if (position >= str.length) break;
+    // Safe to use because we never do special actions for surrogate pairs
+    const char = StringPrototypeCharAt(str, position);
+    let parameterValue = null;
+    if (char === '"') {
+      // Handle quoted-string form of values
+      // skip '"'
+      position += 1;
+      // Find matching closing '"' or end of string
+      //   use $1 to see if we terminated on unmatched '\'
+      //   use $2 to see if we terminated on a matching '"'
+      //   so we can skip the last char in either case
+      const insideMatch = execPattern(
+        QUOTED_VALUE_PATTERN,
+        StringPrototypeSlice(str, position));
+      position += insideMatch[0].length;
+      // Skip including last character if an unmatched '\' or '"' during
+      // unescape
+      const inside = insideMatch[1] || insideMatch[2] ?
+        StringPrototypeSlice(insideMatch[0], 0, -1) :
+        insideMatch[0];
+      // Unescape '\' quoted characters
+      parameterValue = StringPrototypeReplace(inside, QUOTED_CHARACTER, '$1');
+      // If we did have an unmatched '\' add it back to the end
+      if (insideMatch[1]) parameterValue += '\\';
+    } else {
+      // Handle the normal parameter value form
+      const valueEnd = StringPrototypeIndexOf(str, SEMICOLON, position);
+      const rawValue = valueEnd === -1 ?
+        StringPrototypeSlice(str, position) :
+        StringPrototypeSlice(str, position, valueEnd);
+      position += rawValue.length;
+      const trimmedValue = StringPrototypeReplace(rawValue, ENDING_WHITESPACE, '');
+      // Ignore parameters without values
+      if (trimmedValue === '') continue;
```

Comment:
Is it correct that this will skip the increment at the end of the loop when the value is empty but the branch above will not?

Reply:
that is correct, we could add a second branch at the bottom but I don't think perf is really going to be gained by doing the extra check per loop at the bottom.